### PR TITLE
Android Improvements

### PR DESCRIPTION
**Pull request by azdlowry**
- Clean up of tmp on device if specified
- Passing env vars to test binary
- Use of fb-adb if available to fail cargo if tests fail.
- Fix bug with uploading all previous test binaries when using --all

### CODE REVIEW

File: src/android.rs [hunk 1]
```diff
@@ -62,6 +62,9 @@ impl Device for AndroidDevice {
         let bundle_path = exe.parent().ok_or("no parent")?.join("dinghy");
         let bundled_exe_path = bundle_path.join(exe_file_name);
 
+        debug!("Removing previous bundle {:?}", bundle_path);
+        fs::remove_dir_all(&bundle_path).ok();
```
Comment (fredszaq) [inline src/android.rs]: use `let _ = fs::remove_dir_all(&bundle_path)` as it's how it's done in other parts of the code

File: README.md [hunk 2]
```diff
@@ -51,7 +51,8 @@ the fun stuff come.
 
 ## Android setup
 
-You'll need the usual ANDROID_NDK_HOME, and `adb` somewhere in your path.
+You'll need the usual ANDROID_NDK_HOME, and `adb` somewhere in your path. fb-ad
```
Comment (azdlowry) [inline README.md]: Sorry, not sure what you're suggesting here?

Comment (fredszaq) [inline README.md]: you missed the b in fb-adb and you should enclose the executable name with backquotes

**Comment by azdlowry** [comment]
We don't need the changes urgently. Can pull from git repo for now. I may have a look at the ssh changes myself if I can't get adb connection to our devices to be more reliable.

**Pull request merged**